modules/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/io/HasNextTextRecord.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.nativeimpl.io;
+
+import org.ballerinalang.bre.Context;
+import org.ballerinalang.model.types.TypeKind;
+import org.ballerinalang.model.values.BBoolean;
+import org.ballerinalang.model.values.BStruct;
+import org.ballerinalang.model.values.BValue;
+import org.ballerinalang.nativeimpl.io.channels.base.TextRecordChannel;
+import org.ballerinalang.natives.AbstractNativeFunction;
+import org.ballerinalang.natives.annotations.BallerinaFunction;
+import org.ballerinalang.natives.annotations.Receiver;
+import org.ballerinalang.natives.annotations.ReturnType;
+import org.ballerinalang.util.exceptions.BallerinaException;
+
+/**
+ * Native function ballerina.io#hasNextTextRecord.
+ *
+ * @since 0.96.1
+ */
+@BallerinaFunction(
+        packageName = "ballerina.io",
+        functionName = "hasNextTextRecord",
+        receiver = @Receiver(type = TypeKind.STRUCT, structType = "TextRecordChannel", structPackage = "ballerina.io"),
+        returnType = {@ReturnType(type = TypeKind.BOOLEAN)},
+        isPublic = true
+)
+public class HasNextTextRecord extends AbstractNativeFunction {
+    /**
+     * Specifies the index which contains the byte channel in ballerina.lo#hasNextTextRecord.
+     */
+    private static final int TXT_RECORD_CHANNEL_INDEX = 0;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public BValue[] execute(Context context) {
+        BBoolean hasNext;
+        try {
+            BStruct channel = (BStruct) getRefArgument(context, TXT_RECORD_CHANNEL_INDEX);
+            TextRecordChannel textRecordChannel =
+                    (TextRecordChannel) channel.getNativeData(IOConstants.TXT_RECORD_CHANNEL_NAME);
+            hasNext = new BBoolean(textRecordChannel.hasNext());
+        } catch (Throwable e) {
+            String message = "Error occurred while reading text records:" + e.getMessage();
```

Comment:
Can we make sure that the potential throwables which end up here doesn't contain messages that,
1. Doesn't make sense without context (i.e: some exceptions such as `ArrayIndexOutOfBoundsException` contain only the invalid index as the message. The message only makes sense with the name of the exception.)
2. Doesn't expose implementation details (i.e: some exceptions contain Java exception and class names)